test/integration/integration_test.go

```diff
@@ -505,6 +506,13 @@ func (i *integrationTest) exerciseIncrementalBuild(tag, imageName string, remove
 	if checkOnBuild {
 		i.fileExists(containerID, "/sti-fake/src/onbuild")
 	}
+
+	if took := time.Since(start); took > docker.DefaultDockerTimeout {
```

Comment:
I think this approach is worthwhile as long as each expected test run duration is _much_ smaller than DefaultDockerTimeout - otherwise it could lead to flakes.